test/e2e/storage/testsuites/multivolume.go

```diff
@@ -711,15 +699,24 @@ func TestConcurrentAccessToRelatedVolumes(f *framework.Framework, cs clientset.I
 		e2epod.SetAffinity(&node, actualNodeName)
 	}
 
-	// Check that all pods the same content
-	for i, pod := range pods {
-		fileName := "/mnt/volume1/index.html"
-		index := i + 1
+	for _, pvc := range pvcs {
+		var commands []string
 
-		ginkgo.By(fmt.Sprintf("Checking if the volume in pod%d has expected initial content", index))
-		commands := e2evolume.GenerateReadFileCmd(fileName)
-		_, err := framework.LookForStringInPodExec(pod.Namespace, pod.Name, commands, expectedContent, time.Minute)
-		framework.ExpectNoError(err, "failed: finding the contents of the mounted file %s.", fileName)
+		if *pvc.Spec.VolumeMode == v1.PersistentVolumeBlock {
+			fileName := "/mnt/volume1"
+			commands = e2evolume.GenerateReadBlockCmd(fileName, len(expectedContent))
+		} else {
+			fileName := "/mnt/volume1/index.html"
+			commands = e2evolume.GenerateReadFileCmd(fileName)
+		}
+
+		for i, pod := range pods {
+			// Check that all pods have the same content
+			index := i + 1
+			ginkgo.By(fmt.Sprintf("Checking if the volume in pod%d has expected initial content", index))
+			_, err := framework.LookForStringInPodExec(pod.Namespace, pod.Name, commands, expectedContent, time.Minute)
+			framework.ExpectNoError(err, "failed: finding the contents of the mounted file %s.", fileName)
```

Comment:
if there's an error in block it'll be displayed as `mounted file`, probably the if should be inside the loop

Reply:
Ok, good catch Mau, I refactored the output to differentiate between mounted files and block devices, can you please check if it looks better?

Reply:
because the logic is kinda small I'd put everything inside its own if statement, so probably:

```
		if *pvc.Spec.VolumeMode == v1.PersistentVolumeBlock {
			fileName := "/mnt/volume1"
			commands = e2evolume.GenerateReadBlockCmd(fileName, len(expectedContent))
			// Check that all pods have the same content
			index := i + 1
			ginkgo.By(fmt.Sprintf("Checking if the volume in pod%d has expected initial content", index))
			_, err := framework.LookForStringInPodExec(pods[i].Namespace, pods[i].Name, commands, expectedContent, time.Minute)
			framework.ExpectNoError(err, "failed: finding the contents of the block volume %s.", fileName)
		} else {
			fileName := "/mnt/volume1/index.html"
			commands = e2evolume.GenerateReadFileCmd(fileName)
			// Check that all pods have the same content
			index := i + 1
			ginkgo.By(fmt.Sprintf("Checking if the volume in pod%d has expected initial content", index))
			_, err := framework.LookForStringInPodExec(pods[i].Namespace, pods[i].Name, commands, expectedContent, time.Minute)
			framework.ExpectNoError(err, "failed: finding the contents of the mounted file %s.", fileName)
		}
```

Reply:
Ok makes sense Mau! I made the changes to match this